### PR TITLE
WIP: Document the API design discussion decisions from Leysin 2020

### PR DESCRIPTION
**Pull request by hodgestar**

### CODE REVIEW

File: docs/leysin-2020-design-decisions.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+  int size;
+} PointObject;
+
+// When using HPy:
+
+typedef struct {
+    HPyObject_HEAD
+    double x;
+    double y;
+} HPy_Point;
```
Comment (paugier) [inline docs/leysin-2020-design-decisions.md:214]: Just a remark: it seems to me that it's confusing that consumers of the HPy API define variables starting with HPy. Could names starting with HPy be reserved for the HPy API? The equivalent of `PointObject` could be `PointHPy`. At least, it's clear that `PointHPy` is not part of the HPy API but a user defined struct. I don't know if it makes sense?

Comment (antocuni) [reply docs/leysin-2020-design-decisions.md:214]: I did a quick search of `PyType_Ready` on github to check how people are currently calling their C types. A small number of extensions use the `Py` prefix, but the majority seems not to use any particular prefix, so +1 to remove the `HPy_` prefix.

**Comment by timfel** [comment]
Should we merge this? Just thinking if people come by and have some of these questions, it's pretty hard to find right now :)

**Comment by antocuni** [comment]
+1 to merge it even if it's not in a perfect state

File: docs/leysin-2020-design-decisions.md (hunk 1 above)
Comment (hodgestar) [reply docs/leysin-2020-design-decisions.md:214]: Agreed.

**Comment by hodgestar** [comment]
Merging & will make a follow up clean up PR.

**Pull request merged**